redical_core/src/queries/event_query.rs

```diff
@@ -95,6 +100,76 @@ impl<'cal> QueryIndexAccessor<'cal> for EventQueryIndexAccessor<'cal> {
             self.calendar.indexed_class.get_term(&class.to_string())
         )
     }
+
+    fn search_not_uid_index(&self, uid: &str) -> InvertedCalendarIndexTerm {
+        let mut event_uids = self.event_uids.clone();
+
+        event_uids.retain(|event_uid| event_uid != uid);
+
+        InvertedCalendarIndexTerm {
+            events: HashMap::from_iter(
+                event_uids.into_iter().map(|event_uid| 
+                    (event_uid, IndexedConclusion::Include(None)),
+                )
+            ),
+        }
```

Comment:
We can make this more efficient but creating a reference iterator that filters without copying and only copies when needed 🤔 something like:
```suggestion
        let event_uids_iter =
            self.event_uids
                .iter()
                .filter(|event_uid| *event_uid != uid)
                .map(|event_uid| {
                    (event_uid.to_owned(), IndexedConclusion::Include(None))
                });

        InvertedCalendarIndexTerm {
            events: HashMap::from_iter(event_uids_iter),
        }
```